### PR TITLE
Add UI Events as features, update existing events

### PR DESCRIPTION
**Pull request by autonome**
And some tweaks to input and mouse events features.

### CODE REVIEW

File: features/input-event.yml
Comment (ddbeck) [inline features/input-event.yml:1]: I think doing this rename makes some sense, but it's a breaking change. I'd rather not do those one at a time, since we have some other features that ought to be renamed. Would you undo this change and file an issue instead?

(The changes to the keys are fine.)

File: features/focus-events.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Focus events
+description: Focus events represent changes in the focused or blurred elements or windows.
+spec: https://w3c.github.io/uievents/#events-focusevent
+caniuse: focusin-focusout-events
+status:
+  compute_from: api.Element.focus_event
```
Comment (ddbeck) [inline features/focus-events.yml:6]: To align with caniuse, this needs to be pinned to `focusin` and `focusout` but BCD has a very different take on whether or not to mark them supported or not. Would you mind removing all of this and filing an issue in its place, and linking the issue back to this PR discussion?

```suggestion
```

File: features/composition-events.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: Composition events
+description: Composition Events are a method of adding text containing characters that might not be commonly available on keyboard, from emoji to logograms in Asian languages to input from speech recognition processors.
```
Comment (ddbeck) [inline features/composition-events.yml:2]: Would you mind changing the descriptions for all of these to use the convention that we have for other event features?

> [one event] The `foo` event fires when…
> [multiple events] The `bar`, `baz`, and `qux` events fire when…
> [too many events to name] Mouse events, such as `click`, `mousedown`, or `mousemove`, fire when…

I'm guessing you've already familiarized yourself with these and it'll move a lot faster if you seed these conventionally first.

File: features/keyboard-events.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: Keyboard events
+description: Keyboard events describe a single interaction between the user and a key (or combination of a key with modifier keys) on the keyboard.
+spec: https://w3c.github.io/uievents/#events-keyboardevents
+status:
+  compute_from: api.KeyboardEvent.KeyboardEvent
```
Comment (ddbeck) [inline features/keyboard-events.yml:5]: I think this should pin to the `_event` keys

File: features/wheel-events.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+name: Wheel events
+description: Wheel events occur when the user moves a mouse wheel or similar spatially rotating input device.
+spec: https://w3c.github.io/uievents/#events-wheelevents
+status:
+  compute_from: api.WheelEvent
```
Comment (ddbeck) [inline features/wheel-events.yml:5]: Pin to the `_event` keys. If the event doesn't actually fire, there's no sense reporting the status of the interface.

File: features/pointer-events.yml [hunk 5]
```diff
@@ -1,8 +1,13 @@
-draft_date: 2024-11-07
 name: Pointer Events
-description: TODO
+description: Pointer events are an event model to handle pointing input devices such as a mouse, pen/stylus or touch (such as one or more fingers).
 spec: https://w3c.github.io/pointerevents/
+caniuse: pointer
+status:
+  compute_from: api.PointerEvent
```
Comment (ddbeck) [inline features/pointer-events.yml:6]: Pin to `_event` keys only

**Review by ddbeck — CHANGES_REQUESTED** [review]

File: features/pointer-events.yml
Comment (Elchi3) [inline features/pointer-events.yml:1]: needs a new id, as pointer-events is now taken by the CSS property.

File: features/input-event.yml
Comment (autonome) [reply features/input-event.yml:1]: Filed #2324

File: features/focus-events.yml (hunk 1 above)
Comment (autonome) [reply features/focus-events.yml]: Filed #2327

**Comment by autonome** [comment]
Fixed review comments, filed follow up issues and added a comment about the Caniuse vs BCD issue to the focus events feature b/c it kinda stands out oddly being baseline false.

File: features/composition-events.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+name: Composition events
+description: The `compositionstart`, `compositionupdate` and `compositionend` events fire when text containing characters that might not be commonly available on keyboards are added. Examples of these characters are emoji, logograms in Asian languages, or input from speech recognition processors.
```
Comment (ddbeck) [inline features/composition-events.yml:2]: Optional suggestion: Seems like "input method editor" might be a key term here.

```suggestion
description: The `compositionstart`, `compositionupdate`, and `compositionend` events fire when an input method editor adds characters that might not be commonly available on keyboards. Examples of these characters are emoji, logograms in Asian languages, or input from speech recognition processors.
```

File: features/focus-events.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+name: Focus events
+description: The `blur`, `focus`, `focusin` and `focusout` events fire when there are changes in focus for the elements or windows involved.
+spec: https://w3c.github.io/uievents/#events-focusevent
+
+# Caniuse vs BCD is conflicting, so not adding the Caniuse field nor
+# computing status here yet until figured out in this issue:
+#
+# https://github.com/web-platform-dx/web-features/issues/2327
+
```
Comment (ddbeck) [inline features/focus-events.yml:9]: I think we can negotiate our way into a useful-now status, but it's tricky. We have to write the description to dodge the question a bit and compute from `blur` and `focus`.

(Also, I tried really hard to replace "gets or loses focus" with something actually descriptive, but I can't find an actual definition of "focus." A real mystery.)

```suggestion
name: Focus events
description: Focus events, such as `focus` and `blur`, fire when an element gets or loses focus.
spec: https://w3c.github.io/uievents/#events-focusevent
compute_from:
  - api.Element.blur_event
  - api.Element.focus_event
# It's possible that caniuse's `focusin-focusout-events` could be linked here
# or broken out into a separate feature. But caniuse and BCD conflict,
# making this difficult. See this issue to help resolve this:
# https://github.com/web-platform-dx/web-features/issues/2327
```

File: features/keyboard-events.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
+name: Keyboard events
+description: The `keydown` and `keyup` events fire for each interaction between the user and a key (or combination of a key with modifier keys) on the keyboard.
```
Comment (ddbeck) [inline features/keyboard-events.yml:2]:
```suggestion
description: The `keydown` and `keyup` events fire for each key press (or, with modifier keys, a combination of key presses) on a keyboard.
```

File: features/pointer-events-api.yml [hunk 9]
```diff
@@ -1,8 +1,13 @@
-draft_date: 2024-11-07
 name: Pointer Events
-description: TODO
+description: Pointer events are an event model to handle pointing input devices such as a mouse, pen/stylus or touch (such as one or more fingers).
```
Comment (ddbeck) [inline features/pointer-events-api.yml:2]: Tried to work in some entry points and a not-to-be-confused:

```suggestion
description: Pointer events, such as `pointerdown`, and the `PointerEvent` API, represent general pointing inputs, from a wide range of devices, such as a mouse, pen or stylus, or touch with one or more fingers. Not to be confused with the `pointer-events` CSS property.
```

**Review by ddbeck — COMMENTED** [review]
Getting pretty close. Thank you!
